Integrations/GoogleBigQuery/GoogleBigQuery.py

```diff
@@ -0,0 +1,192 @@
+        return_error("Error: priority must have a value of INTERACTIVE or BATCH.")
+    if write_disposition and not (write_disposition == 'WRITE_TRUNCATE' or write_disposition == 'WRITE_APPEND' or write_disposition == 'WRITE_EMPTY'):
+        return_error("Error: write_disposition must have a value of WRITE_TRUNCATE, WRITE_APPEND or WRITE_EMPTY.")
+
+
+def build_query_job_config(allow_large_results, default_dataset_string, destination_table, dry_run, priority, use_query_cache, use_legacy_sql, kms_key_name, write_disposition):
+    validate_args_for_query_job_config(allow_large_results, priority, use_query_cache, use_legacy_sql, dry_run, destination_table, write_disposition)  # send relevant params
+    query_job_config = bigquery.QueryJobConfig()
+    if allow_large_results:
+        query_job_config.allow_large_results = str_to_bool(allow_large_results)
+    if default_dataset_string:
+        query_job_config.default_dataset = default_dataset_string
+    if destination_table:
+        query_job_config.destination = destination_table # must be in the format your-project.your_dataset.your_table
+    if kms_key_name:
+        query_job_config.destination_encryption_configuration = bigquery.table.EncryptionConfiguration(kms_key_name) # make sure this shouldn't be just kms_key_name
+    if dry_run:
+        query_job_config.dry_run = str_to_bool(dry_run)
+    if use_legacy_sql:
+        query_job_config.use_legacy_sql = str_to_bool(use_legacy_sql)
+    if use_query_cache:
+        query_job_config.use_query_cache = str_to_bool(use_query_cache)
+    if priority:
+        query_job_config.priority = priority
+    if write_disposition:
+        query_job_config.write_disposition = write_disposition
+
+    return query_job_config
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+def query(query_string, project_id, location, allow_large_results, default_dataset, destination, kms_key_name, dry_run, priority, use_query_cache, use_legacy_sql,
+          google_service_creds, job_id, write_disposition):
+    bigquery_client = start_and_return_bigquery_client(google_service_creds)
+    job_config = build_query_job_config(allow_large_results, default_dataset, destination, dry_run, priority, use_query_cache, use_legacy_sql, kms_key_name, write_disposition)
+    # not sure if query should be a keyword arg
+    query_job = bigquery_client.query(query = query_string, job_config = job_config, location = location, job_id = job_id, project = project_id)
+    if not (dry_run and str_to_bool(dry_run)):
+        query_results = query_job.result()
+        return query_results
+    # if dry run is activated, the results (number of bytes the query will process) are returned in the job itself
+    else:
+        return query_job
+
+
+def query_command():
+    args = demisto.args()
+    query_results = query(args['query'], args.get('project_id', None), args.get('location', None), args.get('allow_large_results', None),
```

Comment:
If there are default values this will be readable + i prefer you split the `args.get('project_id', None)` to not be in the function call but rather for you to assign params for them beforehand and send the params instead.
And then you can do `query(query=query, project_id=project_id, location=location` or maybe just `query(query, project_id, location`

Reply:
I agree, but Anar prefers it to be done this way (sending the value straight away instead of naming it first).
I'd love to change the concept but I think we need consistency through integrations, whatever the consistent styling decided is.
What do you think?